test/lit/passes/type-merging.wast

```diff
@@ -1052,21 +1052,23 @@
   ;; CHECK:       (type $2 (func (param (ref $A)) (result (ref $B))))
 
   ;; CHECK:      (func $test (type $2) (param $a (ref $A)) (result (ref $B))
-  ;; CHECK-NEXT:  (block $__binaryen_fake_return (result (ref $B))
-  ;; CHECK-NEXT:   (drop
-  ;; CHECK-NEXT:    (br_on_cast $__binaryen_fake_return (ref $A) (ref $B)
-  ;; CHECK-NEXT:     (local.get $a)
-  ;; CHECK-NEXT:    )
-  ;; CHECK-NEXT:   )
-  ;; CHECK-NEXT:   (drop
-  ;; CHECK-NEXT:    (block $l (result (ref $A))
-  ;; CHECK-NEXT:     (br_on_non_null $l
+  ;; CHECK-NEXT:  (block $label (result (ref $B))
+  ;; CHECK-NEXT:   (block (result (ref $B))
```

Comment:
It looks like now we generate two blocks here instead of one. This is probably fine as optimizations will remove the extra one, but just wanted to check this is expected? The thing I'd worry about most here is if extra blocks are added in each roundtrip here.

Reply:
The inner block is the original function body block (which normally would not be printed) and the outer one is a wrapper block for receiving the branch. It looks like the old parser combines these into a single block, which is slightly nicer. There's no risk of roundtrips adding more blocks here since the branches no longer target the function scope.

Reply:
I see, thanks. Would it be hard to reuse the existing function body block in the new parser, as a minor optimization here? Not that important but it would be nice.

Reply:
I can look into it 👍